singer-io/tap-twitter-ads
# TDL-19292 Add new cards stream and removed deprecated streams.

# Description of change
**TDL-19291 Remove deprecated streams and fields** 
- Removed the following deprecated streams,
  -  `Website Cards `
  - `Video Website Cards `
  - `Image App Download Cards `
  - `Video App Download Cards `
  - `Card image direct message`
  - `Video direct message`
- Removed following fields from campaign stream,
  - `start_time`
  - `end_time`

**TDL-19292 Add new field and stream**
- Added [cards](https://developer.twitter.com/en/docs/twitter-ads-api/creatives/api-reference/cards#cards) stream and it's schema.
- Added the `budget_optimization` field into the schema of the campaign stream. 
- Passed all integration test cases in the local environment, that are added in [PR#20](https://github.com/singer-io/tap-twitter-ads/pull/20).


# Manual QA steps
 - Run the discover mode and verify that removed streams are not available in the catalog and the `cards` stream is available in the catalog.
 - Run the sync mode and verify that tap write all records of the `cards` stream. Also, verify that the `cards` stream respects the bookmark key.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
